# general
logging_level: 'INFO' # 10 #

test_option: ""

---
#Data info

dataset_root_dir: '/home/kowalski/data/strl/'
sequence: 'unpacked_10_find'
segmentation_config: '/home/kowalski/data/MIPT_campus/map_aligned_data/anno_config.json'

semantics:
  initial_out_dir: 'init_semantics_conf'
  overwrite_existing_data: False
---
# Semantics 
semantic_uncert_range: 5.
semantic_refine_range: 15.

---

# Sensors
lidar:
  lower_bounds: [-50, -50, -50]
  upper_bounds: [50, 50, 50]
  # base_link
  # base_link2lidar_q: [-0.0078264, 0.0044145, -0.007514, 0.9999314]
  # base_link2lidar_t: [-0.29972749, 0.01367844, 0.88278588]

  #odom
  base_link2lidar_q: [-0.0205339, 0.0651787, -0.4960213, 0.8656171]
  base_link2lidar_t: [-1.26267175, 0.95500783, 2.0047283]


max_ts_delay: 0.05 # 5e-5

# q =  [w, x, y, z]
# t = [x, y, z]
  
front_cam:
  config_path: '/home/kowalski/data/MIPT_campus/calib/zed_calib.yml'
  left:
    lidar2cam_q: [-0.5098432024259881,-0.497602232228181, 0.49791462150659926, -0.4945027372133386]
    lidar2cam_t: [0.04632155672047202, -0.1327858803524132, -0.06092748914769486]

  right:
    lidar2cam_q: [-0.5098432024259881,-0.497602232228181, 0.49791462150659926, -0.4945027372133386] #! Not correct
    lidar2cam_t: [0.04632155672047202, -0.1327858803524132, -0.06092748914769486] #! Not correct
  
back_cam:
  config_path: '/home/kowalski/data/MIPT_campus/calib/realsense_calib.yml'
  left:
    lidar2cam_q: [-0.4996873443841852, -0.4927393159284719, -0.5006918959250938, 0.5067820706029872]
    lidar2cam_t: [0.01827032885440663, -0.14178753281358009, -0.0672336302622617]

  
# lidar2cam_T:
#   - [1, 0, 0, 0]
#   - [0, 1, 0, 0]
#   - [0, 0, 1, 0]


---
#Mesh settings


semantics_mapping:
  unknown:
    labels: [0]
    color: [125, 125, 125]
    voxel_size: 0.3
    ext_weight: 0.3
    bounds: []

  building: 
    labels: [7, 8, 19, 20, 27]
    color: [255, 255, 0]
    voxel_size: 0.2
    ext_weight: 0.3
    bounds: []

  fence: 
    labels: [4, 5, 6, 7, 46, 47, 48, 49, 50, 51, 36, 37, 38, 39, 40, 41, 42, 44, 53]
    color: [255, 127, 0]
    voxel_size: 0.1
    ext_weight: 0.3
    bounds: []

  road:
    labels: [9, 12, 15, 16, 25, 26, 43, 45, 9, 12, 16]
    color: [255, 0, 127]
    voxel_size: 0.15
    ext_weight: 0.3
    bounds: [-7, -50, -50, 7, 50, 50]

  vegetation:
    labels: [31, 32]
    color: [0, 255, 0]
    voxel_size: 0.15
    ext_weight: 0.3
    bounds: []

  snow:
    labels: [30]
    color: [128, 0, 128]
    voxel_size: 0.15
    ext_weight: 0.3
    bounds: []

  moving_object:
    labels: [1, 2, 3, 21, 22, 23, 24, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66]
    color: [0, 0, 255]
    voxel_size: 0.15
    ext_weight: 0.3
    bounds: []

# oneformer
# semantics_mapping:
#   unknown:
#     labels: [0]
#     color: [125, 125, 125]
#     voxel_size: 0.3
#     ext_weight: 3.0

#   building: 
#     labels: [7, 8, 19, 20, 27]
#     color: [230, 230, 0]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   fence: 
#     labels: [4, 5, 6, 7]
#     color: [255, 119, 0]
#     voxel_size: 0.1
#     ext_weight: 3.0

#   road_sign: 
#     labels: [46, 47, 48, 49, 50, 51]
#     color: [250, 170, 30]
#     voxel_size: 0.1
#     ext_weight: 3.0

#   object: 
#     labels: [36, 37, 38, 39, 40, 41, 42, 44, 53]
#     color: [140, 140, 20]
#     voxel_size: 0.15
#     ext_weight: 3.0

#   road:
#     labels: [9, 12, 15, 16, 25, 26, 43, 45]
#     color: [128, 64, 128]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   pedestrian_area:
#     labels: [10, 13, 17]
#     color: [96, 96, 96]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   vegetation:
#     labels: [31, 32]
#     color: [107, 142, 35]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   snow:
#     labels: [30]
#     color: [255, 255, 255]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   moving_object:
#     labels: [1, 2, 3, 21, 22, 23, 24, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66]
#     color: [0, 0, 200]
#     voxel_size: 0.15
#     ext_weight: 3.0

# wo dynamic
# semantics_mapping:
#   unknown:
#     labels: [0]
#     color: [125, 125, 125]
#     voxel_size: 0.3
#     ext_weight: 3.0

#   building: 
#     labels: [6, 7, 18, 19, 26]
#     color: [230, 230, 0]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   fence: 
#     labels: [3, 4, 5, 6]
#     color: [255, 119, 0]
#     voxel_size: 0.1
#     ext_weight: 3.0

#   road_sign: 
#     labels: [45, 46, 47, 48, 49, 50]
#     color: [250, 170, 30]
#     voxel_size: 0.1
#     ext_weight: 3.0

#   object: 
#     labels: [35, 36, 37, 38, 39, 40, 41, 43, 52]
#     color: [140, 140, 20]
#     voxel_size: 0.15
#     ext_weight: 3.0

#   road:
#     labels: [8, 11, 14, 15, 24, 25, 42, 44]
#     color: [128, 64, 128]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   pedestrian_area:
#     labels: [9, 12, 16]
#     color: [96, 96, 96]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   vegetation:
#     labels: [30, 31]
#     color: [107, 142, 35]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   snow:
#     labels: [29]
#     color: [255, 255, 255]
#     voxel_size: 0.2
#     ext_weight: 3.0

#   moving_object:
#     labels: [1, 2, 20, 21, 22, 23, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65]
#     color: [0, 0, 200]
#     voxel_size: 0.15
#     ext_weight: 3.0

# semantics_mapping:
#   classes:
#     unknown: [0]
#     building: [6, 7, 18, 19, 26]
#     fence: [3, 4, 5, 6]
#     road_sign: [45, 46, 47, 48, 49, 50]
#     object: [35, 36, 37, 38, 39, 40, 41, 43, 52]
#     road: [8, 11, 14, 15, 24, 25, 42, 44]
#     pedestrian_area: [9, 12, 16]
#     vegetation: [30, 31]
#     snow: [29]
#     moving_object: [1, 2, 20, 21, 22, 23, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65]
  
#   colors:
#     unknown: [125, 125, 125]
#     building: [70, 70, 70]
#     fence: [255, 119, 0]
#     road_sign: [250, 170, 30]
#     object: [140, 140, 20]
#     road: [128, 64, 128]
#     pedestrian_area: [96, 96, 96]
#     vegetation: [107, 142, 35]
#     snow: [255, 255, 255]
#     moving_object: [0, 0, 200]
